mlrun/model.py

```diff
@@ -1655,7 +1705,12 @@ def _artifact(self, key):
         ]
 
         if not matching_artifacts:
-            return None
+            if key not in self.status.artifact_uris:
+                return None
+
+            # Get artifact by store URI sanity (should have been enriched by now in status.artifacts property)
+            artifact_uri = self.status.artifact_uris[key]
+            matching_artifacts = [mlrun.datastore.get_store_resource(artifact_uri)]
```

Comment:
`mlrun.datastore.get_store_resource(artifact_uri)` can return None, resulting in `matching_artifacts` being a list that contains a single None element.
In this case, you can return the artifact immediately without needing to sort or find non latest artifacts, since it only results in one artifact